optimizely/helpers/condition.py

```diff
@@ -233,12 +239,221 @@ def substring_evaluator(self, index):
+        return self.compare_user_version_with_target_version(index) > 0
+
+    def semver_less_than_evaluator(self, index):
+        """ Evaluate the given semantic version less than match target version for the user version.
+
+      Args:
+        index: Index of the condition to be evaluated.
+
+      Returns:
+        Boolean:
+          - True if the user version is less than the target version.
+          - False if the user version is greater than or equal to the target version.
+        None:
+          - if the user version value is not string type or is null.
+    """
+        return self.compare_user_version_with_target_version(index) < 0
+
+    def semver_less_than_or_equal_evaluator(self, index):
+        """ Evaluate the given semantic version less than or equal to match target version for the user version.
+
+      Args:
+        index: Index of the condition to be evaluated.
+
+      Returns:
+        Boolean:
+          - True if the user version is less than or equal to the target version.
+          - False if the user version is greater than the target version.
+        None:
+          - if the user version value is not string type or is null.
+    """
+        return self.compare_user_version_with_target_version(index) <= 0
+
+    def semver_greater_than_or_equal_evaluator(self, index):
+        """ Evaluate the given semantic version greater than or equal to match target version for the user version.
+
+      Args:
+        index: Index of the condition to be evaluated.
+
+      Returns:
+        Boolean:
+          - True if the user version is greater than or equal to the target version.
+          - False if the user version is less than the target version.
+        None:
+          - if the user version value is not string type or is null.
+    """
+        return self.compare_user_version_with_target_version(index) >= 0
+
+    def split_semantic_version(self, target):
+        """ Method to split the given version.
+
+      Args:
+        target: Given version.
+
+      Returns:
+        List:
+          - The array of version split into smaller parts i.e major, minor, patch etc
+        Exception:
+          - if the given version is invalid in format
+    """
+        target_prefix = target
+        target_suffix = ""
+        target_parts = []
+
+        if self.has_white_space(target):
+            raise Exception(Errors.INVALID_ATTRIBUTE_FORMAT)
+
+        if self.is_pre_release(target):
+            target_parts = target.split(SemverType.IS_PRE_RELEASE)
+        elif self.is_build(target):
+            target_parts = target.split(SemverType.IS_BUILD)
+
+        if target_parts:
+            if len(target_parts) < 1:
+                raise Exception(Errors.INVALID_ATTRIBUTE_FORMAT)
+            target_prefix = str(target_parts[0])
+            target_suffix = target_parts[1:]
+
+        dot_count = target_prefix.count(".")
+        if dot_count > 2:
+            raise Exception(Errors.INVALID_ATTRIBUTE_FORMAT)
```

Comment:
Can you add comments over each of these about what is being checked in the if condition?